CMakeFilters.cmake

```diff
@@ -75,6 +75,8 @@ if (HDF5_ENABLE_Z_LIB_SUPPORT)
       find_package (ZLIB NAMES ${ZLIB_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS static shared)
       if (NOT ZLIB_FOUND)
         find_package (ZLIB) # Legacy find
+        set (H5_ZLIB_INCLUDE_DIR ${ZLIB_INCLUDE_DIRS})
```

Comment:
Don't these also need to be set if the first `find_package` call succeeds?

Reply:
I am confusing two different processes! No wonder it doesn't work

Reply:
Based on this refactoring, it even seems like the `H5_ZLIB_FOUND` and `H5_SZIP_FOUND` aren't really needed at a first glance, though there could be other places they're used that I'm not seeing in the limited diff context.

Reply:
Maybe - but let's make sure everything works first